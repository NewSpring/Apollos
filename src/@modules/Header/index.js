import React, { PureComponent } from 'react';
import {
  Text,
  View,
} from 'react-native';
import PropTypes from 'prop-types';

import createStyleSheet from '@primitives/stylesheet';

const styles = createStyleSheet(({ primaryColor }) => ({
  container: {
    height: 60,
    backgroundColor: primaryColor,
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: 'white',
  },
}));

export default class Header extends PureComponent {
  static propTypes = {
    titleText: PropTypes.string,
  };

  static defaultProps = {
    titleText: '',
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{this.props.titleText}</Text>
      </View>
    );
  }
}
