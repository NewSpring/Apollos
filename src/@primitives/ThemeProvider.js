import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_THEME, THEME_PROPS } from './constants';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    theme: DEFAULT_THEME,
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: Object.assign({}, DEFAULT_THEME, this.props.theme),
  })

  render() {
    return Children.only(this.props.children);
  }
}
