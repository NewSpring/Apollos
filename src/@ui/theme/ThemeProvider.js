import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import createTheme, { THEME_PROPS } from './createTheme';


export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  static defaultProps = {
    children: null,
    themeInput: {},
  };

  static childContextTypes = {
    theme: PropTypes.shape(THEME_PROPS),
    themeInput: PropTypes.shape(THEME_PROPS),
  };

  getChildContext = () => ({
    theme: createTheme(this.props.themeInput),
    themeInput: this.props.themeInput,
  })

  render() {
    return Children.only(this.props.children);
  }
}
