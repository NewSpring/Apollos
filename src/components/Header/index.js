import React, { PureComponent } from 'react';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  container: {
    height: 60,
    backgroundColor: 'green',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    color: 'white',
  },
});

export default class Header extends PureComponent {
  static propTypes = {
    titleText: PropTypes.string,
  };

  static defaultProps = {
    titleText: '',
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{this.props.titleText}</Text>
      </View>
    );
  }
}
