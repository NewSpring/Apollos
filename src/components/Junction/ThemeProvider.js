import { PureComponent, Children } from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_THEME } from './constants';

export default class ThemeProvider extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.shape({
      primaryColor: PropTypes.string,
      secondaryColor: PropTypes.string,
    }),
  };

  static defaultProps = {
    children: null,
    theme: DEFAULT_THEME,
  };

  static childContextTypes = {
    theme: PropTypes.shape({
      primaryColor: PropTypes.string,
      secondaryColor: PropTypes.string,
    }),
  };

  getChildContext = () => ({
    theme: this.props.theme,
  })

  render() {
    return Children.only(this.props.children);
  }
}
