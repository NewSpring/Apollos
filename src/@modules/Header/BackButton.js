import React from 'react';
import { View } from 'react-native';
import { Link } from '@modules/TabBar';
import styled from '@primitives/styled';

const Container = styled({
  position: 'absolute',
  left: 4,
  top: 0,
  bottom: 0,
})(View);

const BackButton = () => (
  <Container>
    <Link pop icon="arrow-back" iconSize={24} />
  </Container>
);

export default BackButton;
