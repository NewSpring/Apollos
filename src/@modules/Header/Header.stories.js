import React from 'react';

import { storiesOf } from '@storybook/react-native';

import Header from '@modules/Header';

storiesOf('Header', module)
  .add('renders', () => <Header />);

