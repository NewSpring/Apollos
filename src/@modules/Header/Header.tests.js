import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '@primitives/ThemeProvider';

import Header from './';

describe('The Header component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Header />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
