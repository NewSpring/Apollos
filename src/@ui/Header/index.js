import React from 'react';
import { View, StatusBar, Platform } from 'react-native';
import SafeAreaView from '@ui/SafeAreaView';
import PropTypes from 'prop-types';
import { compose, setPropTypes, pure, branch, renderNothing, defaultProps } from 'recompose';
import { withTheme, withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';
import { H2, H6 } from '@ui/typography';

import BackButton from './BackButton';

const StyledHeaderBar = styled(
  ({ theme }) => ({
    height: 50,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 25,
    paddingHorizontal: theme.sizing.baseUnit / 2,
    ...Platform.select({
      web: {
        height: undefined,
        paddingHorizontal: theme.sizing.baseUnit,
        paddingVertical: theme.sizing.baseUnit,
        paddingTop: theme.sizing.baseUnit * 2.5,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'flex-start',
      },
    }),
  }),
  'Header.Bar',
)(View);

const HeaderContainer = styled(
  ({ theme, backgroundColor }) => ({
    backgroundColor: backgroundColor || theme.colors.background.primary,
    ...Platform.select({
      android: {
        // paddingTop: 25, // todo: this is currently required as SafeAreaView isn't
        // properly adding padding on android.
      },
      web: {
        backgroundColor: theme.colors.background.paper,
      },
    }),
  }),
  'Header.Container',
)(SafeAreaView);

const WebHeaderText = styled(
  ({ theme }) => ({
    color: theme.colors.darkSecondary,
  }),
  'Header.WebHeaderText',
)(H2);

const StyledHeaderText =
  Platform.OS === 'web'
    ? WebHeaderText
    : styled(
      ({ theme, barStyle }) => ({
        color: barStyle === 'dark-content' ? theme.colors.darkPrimary : theme.colors.lightPrimary,
        maxWidth: '80%',
      }),
      'Header.Text',
    )(H6);

const RightContainer = styled(
  ({ theme }) => ({
    position: 'absolute',
    right: 4,
    top: 0,
    bottom: 0,
    justifyContent: 'center',
    ...Platform.select({
      web: {
        right: theme.sizing.baseUnit,
        top: theme.sizing.baseUnit,
        justifyContent: 'flex-start',
      },
    }),
  }),
  'Header.RightContainer',
)(View);

const ColoredBackButton = compose(
  pure,
  withTheme(({ theme, barstyle }) => ({
    color: barstyle === 'dark-content' ? theme.colors.darkPrimary : undefined,
  })),
)(BackButton);

const enhance = compose(
  defaultProps({
    backButton: false,
    barStyle: 'light-content',
    children: null,
  }),
  setPropTypes({
    webEnabled: PropTypes.bool,
    titleText: PropTypes.string,
    backButton: PropTypes.bool,
    barStyle: PropTypes.oneOf(['light-content', 'dark-content']),
    children: PropTypes.node,
  }),
  branch(({ webEnabled }) => !webEnabled && Platform.OS === 'web', renderNothing),
  branch(
    () => Platform.OS !== 'web',
    withThemeMixin(({ theme, barStyle }) => ({
      type: barStyle === 'light-content' ? 'dark' : 'light',
      colors: {
        background: {
          ...theme.colors.background,
          default: theme.colors.background.primary,
        },
      },
    })),
  ),
  pure,
);

const Header = enhance(
  ({
    titleText,
    right,
    backButton = false,
    barStyle = 'light-content',
    style = {},
    backgroundColor = null,
    children,
  }) => (
    <HeaderContainer backgroundColor={backgroundColor} style={style}>
      <StatusBar barStyle={barStyle} />
      <StyledHeaderBar>
        {backButton ? <ColoredBackButton barStyle={barStyle} /> : null}
        {titleText ? (
          <StyledHeaderText barStyle={barStyle} numberOfLines={1}>
            {titleText}
          </StyledHeaderText>
        ) : null}
        {children}
        {right ? <RightContainer>{right}</RightContainer> : null}
      </StyledHeaderBar>
    </HeaderContainer>
  ),
);

export default Header;
