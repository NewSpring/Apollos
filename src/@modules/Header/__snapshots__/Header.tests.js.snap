// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#6bac43",
      },
      Object {
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    />
  </View>
</View>
`;
